src/net.cpp

```diff
@@ -2883,6 +3013,34 @@ void CConnman::OpenNetworkConnection(const CAddress& addrConnect, bool fCountFai
     }
 }
 
+void CConnman::PrivateBroadcastAdd(size_t n)
+{
+    const size_t before{m_private_broadcast_connections_to_open.fetch_add(n)};
+
+    LogPrintLevel(BCLog::PRIVATE_BROADCAST,
+                  BCLog::Level::Debug,
+                  "Request to increment the needed new connections by %d, changed from %d to %d\n",
+                  n,
+                  before,
+                  before + n);
+}
+
+void CConnman::PrivateBroadcastSub(size_t n)
+{
+    size_t before{m_private_broadcast_connections_to_open.load()};
+    size_t desired;
+    do {
+        desired = before > n ? before - n : 0;
+    } while (!m_private_broadcast_connections_to_open.compare_exchange_weak(before, desired));
```

Comment:
In 6b4693ee859d6bddf559a0b32749b1dafe015ef4 "net: implement opening PRIVATE_BROADCAST connections": Just a question: why using `compare_exchange_weak` instead of `fetch_sub`?

Reply:
To avoid negative values. I added a comment in the source code to explain.